tictactoe/adapter-vdm/src/main/kotlin/com/anaplan/engineering/azuki/tictactoe/adapter/vdm/check/PlayerCanPlaceTokenCheck.kt

```diff
@@ -7,31 +7,26 @@ import com.anaplan.engineering.azuki.tictactoe.adapter.vdm.toVdmPlayer
 import com.anaplan.engineering.azuki.tictactoe.adapter.vdm.toVdmPos
 import com.anaplan.engineering.azuki.vdm.DefaultModuleBuilder
 
-class PlayerCannotPlaceTokenCheck(
-    private val gameName: String,
-    private val playerName: String,
-    private val position: Position
+class PlayerCanPlaceTokenCheck(
+    private val gameName: String, private val playerName: String, private val position: Position, private val expected: Boolean,
 ) : PlaceATokenBehaviour(), DefaultVdmCheck {
 
     override fun build(builder: DefaultModuleBuilder): DefaultModuleBuilder {
         val gameGetter = builder.getters[gameName] ?: throw IllegalStateException("Missing getter for game $gameName")
 
-        return builder.extend(
-            requiredImports = setOf(
-                XOModule.Game.import,
-                XOModule.Pos.import,
-                XOModule.move.import,
-            ),
-            failingStep = """
+        return builder.extend(requiredImports = setOf(
+            XOModule.Game.import,
+            XOModule.Player.import,
+            XOModule.Pos.import,
+            XOModule.isValidMove.import,
+        ), testSteps = listOf("""
                 (
                     dcl g: ${XOModule.Game} := $gameGetter;
+                    dcl p: ${XOModule.Player} := ${toVdmPlayer(playerName)};
                     dcl pos: ${XOModule.Pos} := ${toVdmPos(position)};
-                    let
-                        g' = ${XOModule.move}(g, ${toVdmPlayer(playerName)}, pos)
-                    in
-                        return false
+                    dcl expected: bool := ${if (expected) "true" else "false"};
```

Comment:
⛏️ Doesn't this work?
```suggestion
                    dcl expected: bool := $expected;
```

Reply:
It does (though I forgot to accept the suggestion...)